### ISSUE TITLE
Bug: cannot use theme color name which contains a number after a letter

### ISSUE THREAD

**Issue by serkodev** [unocss/unocss#2600]
### UnoCSS version

0.51.12

### Describe the bug

Cannot use theme color name which contains a number after a letter (like `foo1`, `bar123`, etc.).

But I can use it in Tailwind (sample: https://play.tailwindcss.com/nl5H0hTLL0).

#### Code analysis
Inside [utilities.ts](https://github.com/unocss/unocss/blob/85ca097bb9b3eff7b3333adc37c71726ac4ce671/packages/preset-mini/src/_utils/utilities.ts#L82C1-L84) of `preset-min` code below I think the reason is the regex replace the number after letter.

```js
const colors = main
  .replace(/([a-z])([0-9])/g, '$1-$2')
  .split(/-/g)
```

I am not sure the reason of replacement. Refer to the commit (https://github.com/unocss/unocss/commit/27e7837e1b8cbff90cd8a304a5ab4cc984c267d1) that start to replace I believe the reason is the split the `name` and `no` in the early version of UnoCSS.

Regarding to the code structure currently, I think the replacement is not necessary now?

#### Work around
As currently it will replace `foo1` with `foo-1` so If I define a `1` key inside `foo`, i can use both `foo-1` or `foo1`. But I not sure is it the best prastice.

### Reproduction

```ts
theme: {
  colors: {
    s1: '#ff0000'
  }
}
```

```html
<!-- not work -->
<div class="bg-s1">foo</div>
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/unocss/unocss/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't [already an issue](https://github.com/unocss/unocss/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/unocss/unocss/discussions) or join our [Discord Chat Server](https://chat.antfu.me).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by serkodev** [comment]
If it need to be solved, I can make a PR that remove the regex replacement tho.
Welcome the discuss below. 😃

**Comment by serkodev** [comment]
In case it has to be solved, I made a PR (#2601) that remove the regex replacement tho.
